primitives/src/vid/advz/precomputable.rs

```diff
@@ -0,0 +1,300 @@
+            "(PRECOMPUTE): VID disperse {} payload bytes to {} nodes",
+            payload_byte_len,
+            self.num_storage_nodes
+        ));
+        let chunk_size = self.multiplicity * self.payload_chunk_size;
+        let code_word_size = self.multiplicity * self.num_storage_nodes;
+
+        // partition payload into polynomial coefficients
+        // and count `elems_len` for later
+        let bytes_to_elems = start_timer!(|| "encode payload into field elements");
+        let elems_iter = bytes_to_field::<_, KzgEval<E>>(payload);
+        end_timer!(bytes_to_elems);
+        let inverse_fft = start_timer!(|| "field elements into polynomials (inverse FFT)");
+        let polys: Vec<_> = elems_iter
+            .chunks(chunk_size)
+            .into_iter()
+            .map(|evals_iter| self.polynomial(evals_iter))
+            .collect();
+        end_timer!(inverse_fft);
+
+        // evaluate polynomials
+        let all_storage_node_evals_timer = start_timer!(|| ark_std::format!(
+            "compute all storage node evals for {} polynomials with {} coefficients",
+            polys.len(),
+            chunk_size
+        ));
+        let all_storage_node_evals = {
+            let mut all_storage_node_evals = vec![Vec::with_capacity(polys.len()); code_word_size];
+
+            for poly in polys.iter() {
+                let poly_evals = UnivariateKzgPCS::<E>::multi_open_rou_evals(
+                    poly,
+                    code_word_size,
+                    &self.multi_open_domain,
+                )
+                .map_err(vid)?;
+
+                for (storage_node_evals, poly_eval) in
+                    all_storage_node_evals.iter_mut().zip(poly_evals)
+                {
+                    storage_node_evals.push(poly_eval);
+                }
+            }
+
+            // sanity checks
+            assert_eq!(all_storage_node_evals.len(), code_word_size);
+            for storage_node_evals in all_storage_node_evals.iter() {
+                assert_eq!(storage_node_evals.len(), polys.len());
+            }
+
+            all_storage_node_evals
+        };
+        end_timer!(all_storage_node_evals_timer);
+
+        // vector commitment to polynomial evaluations
+        // TODO why do I need to compute the height of the merkle tree?
+        let all_evals_commit_timer =
+            start_timer!(|| "compute merkle root of all storage node evals");
+        let all_evals_commit =
+            KzgEvalsMerkleTree::<E, H>::from_elems(None, &all_storage_node_evals).map_err(vid)?;
+        end_timer!(all_evals_commit_timer);
+
+        let common_timer = start_timer!(|| ark_std::format!(
+            "(PRECOMPUTE): compute {} KZG commitments",
+            polys.len()
+        ));
+        let common = Common {
+            poly_commits: data.poly_commits.clone(), /* UnivariateKzgPCS::batch_commit(&self.ck,
+                                                      * &polys).map_err(vid)?, */
```

Comment:
```suggestion
            poly_commits: data.poly_commits.clone(),
```
please take out this comment

Reply:
fixed: 09e289a517a3af343c06017d2729df7412b25cdc